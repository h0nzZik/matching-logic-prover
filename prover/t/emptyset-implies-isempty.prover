claim \implies( \and(\equals(variable("S") { Set }, emptyset), .Patterns)
              , \and(isEmpty(variable("S") { Set }, .Patterns), .Patterns)
              )
strategy search-bound(4)
