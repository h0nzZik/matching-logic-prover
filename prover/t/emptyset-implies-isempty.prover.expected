<prover>
  <goal>
    <id>
      root
    </id>
    <active>
      true
    </active>
    <parent>
      .
    </parent>
    <k>
      \implies ( \and ( \equals ( variable ( "S" ) { SetInt } , emptyset ) , .Patterns ) , \exists { .Patterns } \and ( isEmpty ( variable ( "S" ) { SetInt } , .Patterns ) , .Patterns ) )
    </k>
    <strategy>
      success
    </strategy>
    <trace>
      .
    </trace>
  </goal>
</prover>
